pkg/util/nsenter/exec.go

```diff
@@ -0,0 +1,68 @@
+// +build linux
+
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nsenter
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/golang/glog"
+	"k8s.io/utils/exec"
+)
+
+// Executor wraps executor interface to be executed via nsenter
+type Executor struct {
+	// Exec implementation
+	executor exec.Interface
+	// Path to the host filesystem, typically "/rootfs". Used only for testing.
+	hostRootFsPath string
+}
+
+// NewNsenterExecutor returns new nsenter based executor
+func NewNsenterExecutor(hostRootFsPath string, executor exec.Interface) *Executor {
+	nsExecutor := &Executor{
+		hostRootFsPath: hostRootFsPath,
```

Comment:
*rename - hostRootFsPath? Ideally this can be any path right and need not be host root fs right?

Reply:
It has to be path where "/" of host is mounted.

Reply:
If I pass the root of any other container, wouldn't it work? If not this utility library should it ideally be rootnsenter because of the implicit assumptions?

Reply:
Theoretically yes - you can pass root of any other container if you wanted. The problem is, how will mount propagation work from that container into the container which is started for pod?  

So for most practical purposes, we will want this to be host's root.

Reply:
Just the naming is a bit consuming given that this is in util and the assumption doesn't come out clearly. If everyone is fine, I am fine with it. ping @vishh